include/net/nrf_cloud_fota_poll.h

```diff
@@ -30,18 +30,55 @@ enum nrf_cloud_fota_reboot_status {
 };
 
 /**
- * @brief  Event handler registered with the module to handle asynchronous
- * events from the module.
+ * @brief  Reboot event handler registered with the module to handle asynchronous
+ * reboot events from the module.
  *
  * @param[in]  status The reason for the reboot request.
  */
 typedef void (*fota_reboot_handler_t)(enum nrf_cloud_fota_reboot_status status);
 
+/**
+ * @brief  Error event handler registered with the module to handle asynchronous
+ * error events from the module.
+ *
+ * @param[in]  status The FOTA status for the error event.
+ * @param[in]  status_details Details about the error event.
+ */
+typedef void (*fota_error_handler_t)(enum nrf_cloud_fota_status status,
+				     const char *const status_details);
+
 struct nrf_cloud_fota_poll_ctx {
+	/* Internal variables */
 	struct nrf_cloud_rest_context *rest_ctx;
+	struct k_work_delayable timeout_work;
+	bool is_nonblocking;
+	bool full_modem_fota_supported;
 	const char *device_id;
+
+	/* Public variables */
+
+	/** HTTP fragment size. Sets the size of the requested payload in each HTTP request when
+	 *  downloading the firmware image. Note that the header size is not included in this size.
+	 *  It is imporant that the total size of each fragment (HTTP headers + payload) fits in the
+	 *  buffers of the underlying network stack layers. Failing to do so, may result in failure
+	 *  to download new firmware images.
+	 *
+	 *  If this value is set to 0, the Kconfig option
+	 *  :kconfig:option:`CONFIG_NRF_CLOUD_FOTA_DOWNLOAD_FRAGMENT_SIZE` will be used.
+	 */
+	uint32_t fragment_size;
+
+	/** User-provided callback function to handle reboots */
 	fota_reboot_handler_t reboot_fn;
-	bool full_modem_fota_supported;
+
+	/** Optional, user-provided callback function to handle errors.
+	 *  If the function is provided, @ref nrf_cloud_fota_poll_process will be non-blocking and
+	 *  the user will receive error events asynchronously.
+	 *
+	 * If the function is not provided, @ref nrf_cloud_fota_poll_process will be blocking and
+	 * return an error code when an error occurs.
+	 */
+	fota_error_handler_t error_fn;
```

Comment:
if an error does occur is the user responsible for reporting the error status to the cloud?
they would need the job id.

Reply:
I think this should be handled now, I added a subsequent call to `update_job_status()` the places where `error_fn()` is called